gum/gumprocess.h

```diff
@@ -209,6 +209,7 @@ GUM_API gboolean gum_process_resolve_module_pointer (gconstpointer ptr,
     gchar ** path, GumMemoryRange * range);
 GUM_API void gum_process_enumerate_modules (GumFoundModuleFunc func,
     gpointer user_data);
+GUM_API GumModuleDetails * gum_process_get_main_module (void);
```

Comment:
```suggestion
GUM_API const GumModuleDetails * gum_process_get_main_module (void);
```

- Since the main module won't change, it would be good to return a borrowed pointer to the details. We can use the `g_once*()` API to do this lazily and thread-safe.
- Nit: Perhaps we should move it up, just before `resolve_module_pointer()`, as it's the "main" thing, and looking at other modules is going further into the details.

Reply:
done